### PR TITLE
NODE-1297: Option to turn off storage of transactions by address

### PR DESCRIPTION
**Pull request by Karasiq**

### CODE REVIEW

File: src/main/scala/com/wavesplatform/database/Keys.scala [hunk 1]
```diff
@@ -154,4 +156,6 @@ object Keys {
       writeTransactionHN
     )
 
+  val DisableTxsByAddress =
```
Comment (phearnot) [inline src/main/scala/com/wavesplatform/database/Keys.scala]: This option should be configurable from `waves.conf`. There's no need to store this value in LevelDB.

Comment (Karasiq) [reply src/main/scala/com/wavesplatform/database/Keys.scala]: It should be persisted to force user to rebuild the state, if he re-enables transactions by address.

File: src/main/scala/com/wavesplatform/metrics/Metrics.scala [hunk 2]
```diff
@@ -25,6 +28,11 @@ object Metrics extends ScorexLogging {
 
   case class Settings(enable: Boolean, nodeId: Int, influxDb: InfluxDbSettings)
 
+  object Settings {
+    import com.wavesplatform.settings._ // implicit hypherCase
+    implicit val metricsSettingsValueReader: ValueReader[Settings] = arbitraryTypeValueReader
```
Comment (phearnot) [inline src/main/scala/com/wavesplatform/metrics/Metrics.scala]: Please remove all explicit usages of `arbitraryTypeValueReader`, just one import in `WavesSettings.scala` seems to do the trick.

Comment (Karasiq) [reply src/main/scala/com/wavesplatform/metrics/Metrics.scala]: But then a macro reader takes priority and ignores all custom writers from companion objects.

**Pull request merged**